pkg/virt-launcher/notify-client/client.go

```diff
@@ -125,49 +126,65 @@ func libvirtEventCallback(c cli.Connection, domain *api.Domain, event *libvirt.D
 
 	switch domain.Status.Reason {
 	case api.ReasonNonExistent:
-		event := watch.Event{Type: watch.Deleted, Object: domain}
-		client.SendDomainEvent(event)
-		events <- event
+		watchEvent := watch.Event{Type: watch.Deleted, Object: domain}
+		client.SendDomainEvent(watchEvent)
+		events <- watchEvent
 	default:
-		if event.Event == libvirt.DOMAIN_EVENT_DEFINED && libvirt.DomainEventDefinedDetailType(event.Detail) == libvirt.DOMAIN_EVENT_DEFINED_ADDED {
-			event := watch.Event{Type: watch.Added, Object: domain}
-			client.SendDomainEvent(event)
-			events <- event
-		} else if event.Event == libvirt.DOMAIN_EVENT_STARTED && libvirt.DomainEventStartedDetailType(event.Detail) == libvirt.DOMAIN_EVENT_STARTED_MIGRATED {
-			event := watch.Event{Type: watch.Added, Object: domain}
-			client.SendDomainEvent(event)
-			events <- event
-		} else {
-			client.SendDomainEvent(watch.Event{Type: watch.Modified, Object: domain})
+		if libvirtEvent.Event != nil {
+			if libvirtEvent.Event.Event == libvirt.DOMAIN_EVENT_DEFINED && libvirt.DomainEventDefinedDetailType(libvirtEvent.Event.Detail) == libvirt.DOMAIN_EVENT_DEFINED_ADDED {
+				event := watch.Event{Type: watch.Added, Object: domain}
+				client.SendDomainEvent(event)
+				events <- event
+			} else if libvirtEvent.Event.Event == libvirt.DOMAIN_EVENT_STARTED && libvirt.DomainEventStartedDetailType(libvirtEvent.Event.Detail) == libvirt.DOMAIN_EVENT_STARTED_MIGRATED {
+				event := watch.Event{Type: watch.Added, Object: domain}
+				client.SendDomainEvent(event)
+				events <- event
+			} else {
+				client.SendDomainEvent(watch.Event{Type: watch.Modified, Object: domain})
```

Comment:
You'll send the event twice in this case as the same event is sent few lines down for every invocation.